libcxx/test/support/counting_predicates.h

```diff
@@ -41,14 +41,17 @@ struct binary_counting_predicate {
     typedef Arg2 second_argument_type;
     typedef bool result_type;
 
-    binary_counting_predicate ( Predicate p ) : p_(p), count_(0) {}
-    ~binary_counting_predicate() {}
+    TEST_CONSTEXPR_CXX20 binary_counting_predicate(Predicate p) : p_(p), count_(0) {}
+    TEST_CONSTEXPR_CXX20 ~binary_counting_predicate() {}
```

Comment:
Doesn't this work?
```suggestion
    ~binary_counting_predicate() = delete;
```

Reply:
This doesn't work because we need to construct `binary_counting_predicate` objects.

IIUC implicitly declared (defaulted) destructor should works. Although I'm not sure whether it was intended not to use defaulted destructors.

Reply:
Sorry I meant `= default;`, not `= delete;`. The reason why it was not done in the past since `= default` is not part of C++03. Nowadays Clang does support that syntax in C++03.

Reply:
I'm switching to use implicit declared destructor, which is also defaulted and possibly trivial in C++03.